pkg/odo/cli/url/create.go

```diff
@@ -284,8 +351,73 @@ func (o *URLCreateOptions) Run() (err error) {
 					break
 				}
 			}
+			err = o.EnvSpecificInfo.SetConfiguration("url", envinfo.EnvInfoURL{Name: o.urlName, Port: o.componentPort, ExposedPort: o.exposedPort, Kind: o.urlType})
+		} else {
+			containerEndpointMap, err := adapterutils.GetContainerEndpoints(o.devObj.Data)
+			if err != nil {
+				return errors.Wrap(err, "failed to get container endpoint map")
+			}
+			var firstContainer string
+			for containerName, endpointMap := range containerEndpointMap {
+				// save the first container name to store the new endpoint entry
+				if firstContainer == "" {
+					firstContainer = containerName
+				}
+				if _, exist := endpointMap[o.urlName]; exist {
+					if !o.forceFlag && !ui.Proceed(fmt.Sprintf("URL %v already exist in devfile endpoint entry under container %v. Do you want to override the endpoint", o.urlName, containerName)) {
+						log.Info("Aborted by the user")
+						return nil
+					}
+					if len(o.container) > 0 && o.container != containerName {
+						delete(endpointMap, o.urlName)
+					}
+					containerEndpointMap[containerName] = endpointMap
+				}
+				for _, endpoint := range endpointMap {
+					if int(endpoint.TargetPort) == o.componentPort {
+						// devfile cannot have two endpoints within different containers share the same targetport
+						// it is because containers in a single pod shares the network namespace
+						if len(o.container) > 0 && o.container != containerName {
+							return fmt.Errorf("cannot set URL %v under container %v, TargetPort %v is being used for endpoint %v under container %v", o.urlName, o.container, o.componentPort, endpoint.Name, containerName)
+						} else {
+							o.container = containerName
+						}
+						break
+					}
+				}
+			}
+			if reflect.DeepEqual(containerEndpointMap, map[string]map[string]common.Endpoint{}) {
+				// devfile have no containers had endpoint entry
```

Comment:
`have no containers and entrypoint`? or `with entrypoint`?

Reply:
updated